### ISSUE TITLE
folding02 test fails on OpenBSD

### ISSUE THREAD

**Issue by ibara** [flang-compiler/f18#482]
Hi --

Not sure if this is known or expected, but test 12, folding02, fails on OpenBSD. I ran test_folding.sh manually (with #481 applied) and got the following output:
```
 $ ./test_folding.sh folding02.f90                                                                                                              
Assuming no libpgmath support
folding test failed:

actual at 129: in a macro defined here
actual at 129: in a macro defined here
actual at 162: invalid argument on folding function with host runtime
actual at 164: invalid argument on folding function with host runtime
FAIL
```
f18 is indeed built without libpgmath (but I could build with if that matters).

Thanks!

**Comment by jeanPerier** [comment]
Hi,
Thanks for reporting the issue. Are you building f18 with clang and libc++ ? If so, I think I know what the issue is. I will try to push a fix asap.

**Comment by ibara** [comment]
Yes. clang-7.0.1 and libc++.

**Comment by jeanPerier** [comment]
This issue should be fixed by PR #483. There was an issue handling floating point environment during constant folding when f18 was compiled with clang++ that showed-up only in release mode when using libc++.
I do not think OpenBSD plays a role here because I could reproduce it under Linux, but I could not check that #483 works fine on OpenBSD, so I would be glad if you could.

**Comment by ibara** [comment]
It is fixed now, thanks!
